### DISCUSSION TITLE
Can I modify structure returned by xxx_unpack() functions?

### DISCUSSION THREAD

**Discussion by Saigut** [protobuf-c/protobuf-c#623] [Q&A] [answered]
xxx_unpack function will unpack a serialized message into an in-memory structure representation.

Can I modify that structure? 
If I modified that structure, dose xxx_free_unpacked function still work as expected?

For example, xxx_unpack return a structure xxx_stuct, and I modify the value:
xxx_stuct->pointer1 = NULL;
xxx_stuct->pointer2 = some_thing;
xxx_stuct->value = 123;
And then dose xxx_free_unpacked function still work as expected?

**Comment by Saigut** [comment]
Is there anyone can answer my question?

**Comment by lipnitsk** [comment, marked as answer]
You can, but as with everything in C, you have to be careful with memory management. There is no other structure that holds references to allocated memory, so if you overwrite a pointer without first freeing it you will have a memory leak in your code.
